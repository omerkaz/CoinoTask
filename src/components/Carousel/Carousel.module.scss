@import "../../assets/styles/_variables.scss";
.carousel {
  border-radius: 7px 7px 7px 7px;
  box-shadow: rgba(0, 0, 0, 0.17) 0px -23px 25px 0px inset, rgba(0, 0, 0, 0.15) 0px -36px 30px 0px inset, rgba(0, 0, 0, 0.1) 0px -79px 40px 0px inset, rgba(0, 0, 0, 0.06) 0px 2px 1px, rgba(0, 0, 0, 0.09) 0px 4px 2px, rgba(0, 0, 0, 0.09) 0px 8px 4px, rgba(0, 0, 0, 0.09) 0px 16px 8px, rgba(0, 0, 0, 0.09) 0px 32px 16px;
  background-color: $primary-color;
  padding-top: 2rem;
  padding-bottom: 1rem;
}
.image img {
  height: 400px;
  width: 700px;
  object-fit: scale-down;
}
.navigation {
  display: flex;
  justify-content: center;
  margin-top: 1rem;
  padding: 10px;
  background-color: $thirdy-color;
  .button {
    background-color: unset;
    border: 0;
    color: $primary-text-color;
    border-radius: 4px;
  }
}
