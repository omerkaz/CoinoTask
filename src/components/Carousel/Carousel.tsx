import React, { useState, useEffect } from "react";
import styles from "./Carousel.module.scss";
import { FaArrowCircleRight, FaArrowCircleLeft } from "react-icons/fa";
type CaurouselProps = {
  images: { id: number; src: string }[];
  text?: string;
};

function Carousel({ images, text }: CaurouselProps) {
  const [currentImage, setCurrentImage] = useState(0);

  useEffect(() => {
    const intervalId = setInterval(() => {
      setCurrentImage((prevValue) =>
        prevValue === images.length - 1 ? 0 : prevValue + 1
      );
    }, 3000);
    return () => clearInterval(intervalId);
  }, [images.length]);

  const handlePrevClick = () => {
    setCurrentImage(currentImage === 0 ? images.length - 1 : currentImage - 1);
  };

  const handleNextClick = () => {
    setCurrentImage(currentImage === images.length - 1 ? 0 : currentImage + 1);
  };
  return (
    <div className={styles.carousel}>
      <div className={styles.image}>
        <img src={images[currentImage].src} />
        {text ? <div>Text</div> : undefined}
      </div>
      <div className={styles.navigation}>
        <button className={styles.button} onClick={handleNextClick}>
          <FaArrowCircleLeft size={"30"} />
        </button>
        <button className={styles.button} onClick={handlePrevClick}>
          <FaArrowCircleRight size={"30"} />
        </button>
      </div>
    </div>
  );
}

export default Carousel;
