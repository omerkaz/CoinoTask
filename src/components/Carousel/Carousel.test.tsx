import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Carousel from "@src/components/Carousel/Carousel";
describe("Carousel elements renders correctly", () => {
  test("snapshot test", () => {
    // arrange
    jest.useFakeTimers();
    const images = [
      { id: 0, src: "image1.png" },
      { id: 1, src: "image2.png" },
    ];
    const componentTree = render(<Carousel images={images} />);
    expect(componentTree).toMatchSnapshot();
  });

  test("test image", () => {
    // arrange
    const images = [
      { id: 0, src: "image1.png" },
      { id: 1, src: "image2.png" },
    ];
    render(<Carousel images={images} />);
    const carouselImage = screen.getByTestId("carouselImage");
    expect(carouselImage).toBeInTheDocument();
  });
  test("should show the next image when the right button is clicked", () => {
    // arrange
    jest.useFakeTimers();
    const images = [
      { id: 0, src: "image1.png" },
      { id: 1, src: "image2.png" },
    ];
    render(<Carousel images={images} />);
    const rightButton = screen.getByTestId("carouselButtonRight");
    const image = screen.getByTestId("carouselImage");

    // act
    userEvent.click(rightButton);

    // assert
    expect(image).toHaveAttribute("src", "image2.png");
  });

  test("should show the previous image when the left button is clicked", () => {
    // arrange
    jest.useFakeTimers();
    const images = [
      { id: 0, src: "image1.png" },
      { id: 1, src: "image2.png" },
    ];
    render(<Carousel images={images} />);
    const leftButton = screen.getByTestId("carouselButtonLeft");
    const image = screen.getByTestId("carouselImage");

    // act
    userEvent.click(leftButton);

    // assert
    expect(image).toHaveAttribute("src", "image2.png");
  });
});
