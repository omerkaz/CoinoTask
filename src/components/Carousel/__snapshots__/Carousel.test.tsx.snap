// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel elements renders correctly snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div
          class="image"
        >
          <img
            data-testid="carouselImage"
            src="image1.png"
          />
        </div>
        <div
          class="navigation"
        >
          <button
            class="button"
            data-testid="carouselButtonLeft"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
              />
            </svg>
          </button>
          <button
            class="button"
            data-testid="carouselButtonRight"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div
        class="image"
      >
        <img
          data-testid="carouselImage"
          src="image1.png"
        />
      </div>
      <div
        class="navigation"
      >
        <button
          class="button"
          data-testid="carouselButtonLeft"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
            />
          </svg>
        </button>
        <button
          class="button"
          data-testid="carouselButtonRight"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
